gp-advanced-select/gpadvs-preserve-choice-order.php

```diff
@@ -22,10 +22,17 @@
 	$value_array = json_decode( $value, true );
 
 	// Create a map of choice values to their respective index, and sort with that.
-	$value_indices = array_flip( array_column( $field->choices, 'value' ) );
-	usort( $value_array, function ( $x, $y ) use ( $value_indices ) {
-		return $value_indices[ $x ] - $value_indices[ $y ];
-	});
+	if ( is_array( $field->choices ) && ! empty( $field->choices ) && $value_array ) {
+		$valid_values = array_filter( array_column( $field->choices, 'value' ), function ( $value ) {
+			return is_string( $value ) || is_int( $value );
+		});
+		$value_indices = array_flip( $valid_values );
+		if ( is_array( $value_indices ) ) {
+			usort( $value_array, function ( $x, $y ) use ( $value_indices ) {
+				return $value_indices[ $x ] - $value_indices[ $y ];
+			});
+		}
+	}
```

Comment:
```suggestion
	if ( is_array( $field->choices ) && ! empty( $field->choices ) && $value_array ) {
		// Avoid PHP notices for non-string, non-integer values with `array_flip()` by casting everything to a string and ensuring the result is an array.
		$value_indices = (array) array_flip( array_map( 'strval', array_column( $field->choices, 'value' ) ) );
		usort( $value_array, function ( $x, $y ) use ( $value_indices ) {
			return $value_indices[ $x ] - $value_indices[ $y ];
		} );
	}
```

Thoughts on simplifying the code to something like this? Also added a comment about why we're gettin' fancy.

Reply:
Confirmed by. customer here: https://secure.helpscout.net/conversation/2522300498/62231?folderId=7098280#thread-7632446291